src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -327,46 +351,52 @@ bool CanIgnoreAllCollationElements(const UCollator* pColl, const UChar* lpStr, i
 
 }
 
-extern "C" ResultCode GlobalizationNative_GetSortHandle(const char* lpLocaleName, SortHandle** ppSortHandle)
+ResultCode GlobalizationNative_GetSortHandle(const char* lpLocaleName, SortHandle** ppSortHandle)
 {
-    assert(ppSortHandle != nullptr);
-    
-    *ppSortHandle = new (std::nothrow) SortHandle();
-    if ((*ppSortHandle) == nullptr)
+    assert(ppSortHandle != NULL);
+
+    *ppSortHandle = (SortHandle*)malloc(sizeof(SortHandle));
```

Comment:
It would be nice to encapsulate the whole creation (including the pthread_mutex_init) to a separate function, say CreateSortHandle or something like that. It is not obvious this way that the mutex stuff is part of the initialization.